samples/Rewrite/ConditionParser/ConditionRegexParser.cs

```diff
@@ -0,0 +1,162 @@
+
+            if (context.Current == Not)
+            {
+                results.Invert = true;
+                if (!context.Next())
+                {
+                    return null;
+                }
+            }
+
+            if (context.Current == Greater)
+            {
+                if (!context.Next())
+                {
+                    throw new FormatException(); 
+                }
+                if (context.Current == EqualSign)
+                {
+                    if (!context.Next())
+                    {
+                        throw new FormatException();
+                    }
+                    results.Operation = OperationType.GreaterEqual;
+                    results.Type = ConditionType.StringComp;
+                } 
+                else
+                {
+                    results.Operation = OperationType.Greater;
+                    results.Type = ConditionType.StringComp;
+                }
+            }
+            else if (context.Current == Less)
+            {
+                if (!context.Next())
+                {
+                    throw new FormatException();
+                }
+                if (context.Current == EqualSign)
+                {
+                    if (!context.Next())
+                    {
+                        throw new FormatException();
+                    }
+                    results.Operation = OperationType.LessEqual;
+                    results.Type = ConditionType.StringComp;
+                }
+                else
+                {
+                    results.Operation = OperationType.Less;
+                    results.Type = ConditionType.StringComp;
+                }
+            }
+            else if (context.Current == EqualSign)
+            { 
+                if (!context.Next())
+                {
+                    throw new FormatException();
+                }
+                results.Operation = OperationType.Equal;
+                results.Type = ConditionType.StringComp;
+            }
+            else if (context.Current == Dash)
+            {
+                results = ParseAttributeTest(context);
+                if (results.Type == ConditionType.FileTest)
+                {
+                    return results;
+                }
+                context.Next();
+            }
+            else
+            {
+                results.Type = ConditionType.Regex; 
+            }
+
+            context.Mark();
+            while (context.Next()) ;
+            results.Variable = context.Capture();
+            return results;
+        }
+
+        public static GeneralExpression ParseAttributeTest(ConditionParserContext context)
+        {
+            var current = String.Empty;
+            while (true)
+            {
+                if (!context.Next())
+                {
+                    throw new FormatException();
+                }
+                current = String.Concat(current, context.Current);
```

Comment:
Actually, you don't need this at all. Just look at the current character and the one following using the context. Then replace the switch with a set of if statements